src/main.rs

```diff
@@ -82,26 +81,15 @@ enum Commands {
 // Looking at https://unix.stackexchange.com/questions/331611/do-progress-reports-logging-information-belong-on-stderr-or-stdout
 // it is a little complicated.
 // The current setup is to write all progress information to STDERR and all data to STDOUT.
-
-struct OutWriter {
-    stderr: Mutex<tokio::io::Stderr>,
-}
-
-impl OutWriter {
-    pub fn new() -> Self {
-        let stderr = tokio::io::stderr();
-        Self {
-            stderr: Mutex::new(stderr),
-        }
-    }
-}
-
-impl OutWriter {
-    pub async fn println(&self, mut content: String) {
-        let stderr = &mut *self.stderr.lock().await;
-        content.push('\n');
-        stderr.write_all(content.as_bytes()).await.unwrap();
-    }
+macro_rules! progress {
+    // Match a format string followed by any number of arguments
+    ($fmt:expr $(, $args:expr)*) => {{
+        // Use the `format!` macro to format the string with the arguments
+        let mut message = format!($fmt $(, $args)*);
+        // Print the formatted string to the console with a newline
+        message.push('\n');
+        tokio::io::stderr().write_all(message.as_ref()).await.unwrap();
+    }};
```

Comment:
So why even use our own macro over `println!`?  Should we not call `println!` and `eprintln!` macros at all in tokio code?  My gut feeling says this only matters when we're piping entire files in and out of stdio/stdout and not for printing progress.  We already call `println!` in main anyway.

The benefit of directly using `eprintln!` is that we don't even have to worry about this garbled output problem as the stdlib takes care of it.

I admit wrapping `eprintln!` into `progress!` is nice sematically as it's more obvious what we're printing.

Reply:
I think the difference is that `eprintln!` can block, and this would cause havoc if there is only a single thread. I mean, this is the reason tokio::io::stderr() exists I guess...